### DISCUSSION TITLE
Is it possible to capture the status code when using gh cli to GET a branch?

### DISCUSSION THREAD

**Discussion by rnpaul** [community/community#41763] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

I'm attempting to rename branches in a large number of repos, which on its own is easy enough. However, some of the repos don't currently contain the branch I'm renaming (master), in which case the script fails and exits. I would like to check each one to see if the master branch currently exists first, so is there a way to capture the status code after [getting the branch](https://docs.github.com/en/rest/branches/branches?apiVersion=2022-11-28#get-a-branch) with the gh rest api?

**Comment by airtower-luna** [comment]
If by "gh rest api" you mean that you're using the `gh api` command, you can add the `-i` parameter:

```
gh api -i /repos/OWNER/REPO/branches/master
```

It'll add status line and headers to the output, so you could `grep` for `404 Not Found` or similar. The catch is that if you then want to process the JSON payload you have to split headers from payload in the output (at the first empty line). It might be simpler to check if the response object contains a `message` property with the value `Branch not found`. :wink: 

If you use any other HTTP client the way to do it depends on the client you use.

**Reply by rnpaul** [reply]
That makes sense, but I'm unsure how to check the message property in a Bash script.

**Reply by airtower-luna** [reply]
The same way as any other JSON data. I'd use `jq`:

```sh
if [ "$(gh api /repos/OWNER/REPO/branches/master | jq -r .message)" = "Branch not found" ]; then
    echo "branch doesn't exist"
fi
```

Obviously you'll need adjust what to do/not to do, but that's the general idea.
